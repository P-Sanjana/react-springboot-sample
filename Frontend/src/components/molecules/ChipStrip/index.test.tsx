import React from "react";

import {
  render,
  RenderResult,
  fireEvent,
  screen,
} from "@testing-library/react";
import CurrencyChipStrip, { CurrencyChipData } from ".";

const testDataStatic: Array<CurrencyChipData> = [
  {
    name: "Bitcoin",
    color: "#F7931A",
    selected: true,
  },
  {
    name: "XRP",
    color: "#222222",
  },
  {
    name: "Polkadot",
    color: "#E6007A",
  },
  {
    name: "Ethereum",
    color: "#627EEA",
  },
  {
    name: "Tether",
    color: "#26A17B",
  },
  {
    name: "Ethereum 2",
    color: "#191971",
  },
  {
    name: "Dodge Coin",
    color: "#DBC984",
  },
];

describe("CurrencyChipStrip molecule component", () => {
  it("should match snapshot", () => {
    const testData = testDataStatic.slice();
    const onClick = (index: number) => {
      console.log(`Selected an item ${index}`);
    };
    const component = render(
      <CurrencyChipStrip data={testData} onClick={onClick} />
    );
    expect(component).toMatchSnapshot();
  });
  it("should highlight the selected", () => {
    let testData = testDataStatic.slice();
    let component: RenderResult | undefined = undefined;
    const onClick = (index: number) => {
      testData = testDataStatic.slice();
      testData[index].selected = !testData[index].selected;
      if (component) {
        component.rerender(
          <CurrencyChipStrip data={testData} onClick={onClick} />
        );
      }
    };
    component = render(<CurrencyChipStrip data={testData} onClick={onClick} />);
    const chipItem = screen.getByTestId(1);
    fireEvent.click(chipItem);
    expect(chipItem).toHaveStyle({
      border: "2px solid",
    });
  });
  it("should scroll when arrows clicked", async () => {
    const testData = testDataStatic.slice();
    const onClick = (index: number) => {
      console.log(`Selected an item ${index}`);
    };
    render(
      <div style={{ width: "400px" }}>
        <CurrencyChipStrip data={testData} onClick={onClick} />
      </div>
    );
    fireEvent.click(screen.getByTestId("scroll-left-button").childNodes[0]);
    await new Promise((r) => setTimeout(r, 1000));
    fireEvent.click(screen.getByTestId("scroll-right-button").childNodes[0]);
    await new Promise((r) => setTimeout(r, 1000));
  });
});
