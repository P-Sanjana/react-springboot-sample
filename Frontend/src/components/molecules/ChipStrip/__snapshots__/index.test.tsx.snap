// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyChipStrip molecule component should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root MuiBox-root-7 makeStyles-root-1"
      >
        <div
          class="MuiBox-root MuiBox-root-8"
          data-testid="scroll-left-button"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <img
                alt=""
                src="chevron-left.svg"
              />
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root MuiBox-root-9 makeStyles-scrollableContainer-2"
        >
          <div
            class="MuiBox-root MuiBox-root-10 makeStyles-stripContainer-3"
          >
            <button
              class="MuiButtonBase-root makeStyles-chipBoxBase-4"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiBox-root MuiBox-root-11 makeStyles-chipBox-5 makeStyles-chipBoxSelected-6"
                data-testid="0"
                style="background-color: rgba(247, 147, 26, 0.2); border-color: #f7931a;"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2"
                >
                  Bitcoin
                </p>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root makeStyles-chipBoxBase-4"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiBox-root MuiBox-root-12 makeStyles-chipBox-5"
                data-testid="1"
                style="background-color: rgba(34, 34, 34, 0.2);"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2"
                >
                  XRP
                </p>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root makeStyles-chipBoxBase-4"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiBox-root MuiBox-root-13 makeStyles-chipBox-5"
                data-testid="2"
                style="background-color: rgba(230, 0, 122, 0.2);"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2"
                >
                  Polkadot
                </p>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root makeStyles-chipBoxBase-4"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiBox-root MuiBox-root-14 makeStyles-chipBox-5"
                data-testid="3"
                style="background-color: rgba(98, 126, 234, 0.2);"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2"
                >
                  Ethereum
                </p>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root makeStyles-chipBoxBase-4"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiBox-root MuiBox-root-15 makeStyles-chipBox-5"
                data-testid="4"
                style="background-color: rgba(38, 161, 123, 0.2);"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2"
                >
                  Tether
                </p>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root makeStyles-chipBoxBase-4"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiBox-root MuiBox-root-16 makeStyles-chipBox-5"
                data-testid="5"
                style="background-color: rgba(25, 25, 113, 0.2);"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2"
                >
                  Ethereum 2
                </p>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root makeStyles-chipBoxBase-4"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiBox-root MuiBox-root-17 makeStyles-chipBox-5"
                data-testid="6"
                style="background-color: rgba(219, 201, 132, 0.2);"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2"
                >
                  Dodge Coin
                </p>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-18"
          data-testid="scroll-right-button"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <img
                alt=""
                src="chevron-right.svg"
              />
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root MuiBox-root-7 makeStyles-root-1"
    >
      <div
        class="MuiBox-root MuiBox-root-8"
        data-testid="scroll-left-button"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <img
              alt=""
              src="chevron-left.svg"
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root MuiBox-root-9 makeStyles-scrollableContainer-2"
      >
        <div
          class="MuiBox-root MuiBox-root-10 makeStyles-stripContainer-3"
        >
          <button
            class="MuiButtonBase-root makeStyles-chipBoxBase-4"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiBox-root MuiBox-root-11 makeStyles-chipBox-5 makeStyles-chipBoxSelected-6"
              data-testid="0"
              style="background-color: rgba(247, 147, 26, 0.2); border-color: #f7931a;"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                Bitcoin
              </p>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root makeStyles-chipBoxBase-4"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiBox-root MuiBox-root-12 makeStyles-chipBox-5"
              data-testid="1"
              style="background-color: rgba(34, 34, 34, 0.2);"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                XRP
              </p>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root makeStyles-chipBoxBase-4"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiBox-root MuiBox-root-13 makeStyles-chipBox-5"
              data-testid="2"
              style="background-color: rgba(230, 0, 122, 0.2);"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                Polkadot
              </p>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root makeStyles-chipBoxBase-4"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiBox-root MuiBox-root-14 makeStyles-chipBox-5"
              data-testid="3"
              style="background-color: rgba(98, 126, 234, 0.2);"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                Ethereum
              </p>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root makeStyles-chipBoxBase-4"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiBox-root MuiBox-root-15 makeStyles-chipBox-5"
              data-testid="4"
              style="background-color: rgba(38, 161, 123, 0.2);"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                Tether
              </p>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root makeStyles-chipBoxBase-4"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiBox-root MuiBox-root-16 makeStyles-chipBox-5"
              data-testid="5"
              style="background-color: rgba(25, 25, 113, 0.2);"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                Ethereum 2
              </p>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root makeStyles-chipBoxBase-4"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiBox-root MuiBox-root-17 makeStyles-chipBox-5"
              data-testid="6"
              style="background-color: rgba(219, 201, 132, 0.2);"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                Dodge Coin
              </p>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-18"
        data-testid="scroll-right-button"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <img
              alt=""
              src="chevron-right.svg"
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
